### DISCUSSION TITLE
How to parse JSON body when content type is text/plain

### DISCUSSION THREAD

**Discussion by ghost** [actix/actix-web#2319] [Q&A] [answered]
Hi, I have a similar request as #635.
A [PR](https://github.com/actix/actix-web/pull/637) had been merged, but still I got the error.

related code:
```rust
let mut response = client::Client::default() ‣mut response: ClientResponse<Decoder<Payload>> ‣Client
        .get(this_req) ‣ClientRequest
        .header("User-Agent", "Actix-web") ‣ClientRequest
        .send() ‣SendClientRequest
        .await?;
let data = response.json().await?;
log::info!("{:#?}", data);
```

the error:
```bash
curl -X POST -H "Content-Type: application/json" --url http://127.0.0.1:8080/xx -d '{"code":"xxx"}'
# normally it should be (in bytes format)
[2021-07-10T05:41:51Z INFO  x::http] "{\"errcode\":41129,\"errmsg\":\"invalid code, hints: [ req_id: xxx ]\"}"
# but instead
[2021-07-10T05:56:26Z ERROR actix_http::response] Internal Server Error: ContentType
```

I also noticed the newest paradox text in [documentation](https://docs.rs/actix-web/3.3.2/actix_web/client/struct.ClientResponse.html#method.json).
But what is the ultimate solution?
Shall I drop actix and use other crate to achieve such a basic feature?

**Comment by robjtede** [comment, marked as answer]
Just get the body bytes with .body instead of .json and use serde_json::from_slice.
